src/analysis/processing/qgsalgorithmshortestline.cpp

```diff
@@ -0,0 +1,180 @@
+                      "destination layer is taken into account. "
+                      "The n-nearest neighboring features number can be specified. "
+                      "If a maximum distance is specified, then only "
+                      "features which are closer than this distance will "
+                      "be considered. The output features will contain all the "
+                      "source layer attributes, all the attributes from the n-nearest "
+                      "feature and the additional field of the distance. "
+                      "This algorithm uses purely Cartesian calculations for distance, "
+                      "and does not consider geodetic or ellipsoid properties when "
+                      "determining feature proximity."
+                    );
+}
+
+QgsShortestLineAlgorithm *QgsShortestLineAlgorithm::createInstance() const
+{
+  return new QgsShortestLineAlgorithm();
+}
+
+void QgsShortestLineAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "SOURCE" ), QObject::tr( "Source layer" ), QList<int>() << QgsProcessing::TypeVectorAnyGeometry ) );
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "DESTINATION" ), QObject::tr( "Destination layer" ), QList<int>() << QgsProcessing::TypeVectorAnyGeometry ) );
+  addParameter( new QgsProcessingParameterEnum( QStringLiteral( "METHOD" ), QObject::tr( "Method" ), QStringList() << "distance to nearest point on geometry" << "distance to geometry centroid", false, QVariant( "boundary" ) ) );
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "NEIGHBORS" ), QObject::tr( "Maximum number of neighbors" ), QgsProcessingParameterNumber::Integer, 1, false, 1 ) );
+  addParameter( new QgsProcessingParameterDistance( QStringLiteral( "DISTANCE" ), QObject::tr( "Maximum distance" ), QVariant(), QString( "SOURCE" ), true ) );
+  addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "OUTPUT" ), QObject::tr( "Shortest lines" ) ) );
+}
+
+bool QgsShortestLineAlgorithm::prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback * )
+{
+  mSource.reset( parameterAsSource( parameters, QStringLiteral( "SOURCE" ), context ) );
+  if ( !mSource )
+    throw QgsProcessingException( invalidSourceError( parameters, QStringLiteral( "SOURCE" ) ) );
+
+  mDestination.reset( parameterAsSource( parameters, QStringLiteral( "DESTINATION" ), context ) );
+  if ( !mSource )
+    throw QgsProcessingException( invalidSourceError( parameters, QStringLiteral( "DESTINATION" ) ) );
+
+  mMethod = parameterAsInt( parameters, QStringLiteral( "METHOD" ), context );
+
+  mKNeighbors = parameterAsInt( parameters, QStringLiteral( "NEIGHBORS" ), context );
+
+  double paramMaxDist = parameterAsDouble( parameters, QStringLiteral( "DISTANCE" ), context ); //defaults to zero if not set
+  if ( paramMaxDist )
+    mMaxDistance = paramMaxDist;
```

Comment:
```suggestion
  mMaxDistance = parameterAsDouble( parameters, QStringLiteral( "DISTANCE" ), context ); //defaults to zero if not set
```

It's the exact same outcome :smile: